### ISSUE TITLE
KSP refuses to load when it gets to BDArmory full set/BDArmory/Parts/20mmVulcan/vulcanTurret/bahaGatlingGun.

### ISSUE THREAD

**Issue by John-Adler** [PapaJoesSoup/BDArmory#812]
When reporting issues please provide the following information:

[See this for more information on reporting issues for KSP Mods](http://forum.kerbalspaceprogram.com/?showtopic=83212)

[See the Original Forum Post (OP) for additional info](http://forum.kerbalspaceprogram.com/?showtopic=155014)

##### ISSUE TYPE

 - Bug Report   (Something is broken)

##### KSP and BDA Version
KSP version 1.12.3
```
BDA version 1.3.5_beta
```

##### OS / ENVIRONMENT
Windows 10

##### SUMMARY

KSP stops loading at.... well, the thing's in the title. I'm not typing that all again lol.
##### Log Files

And here's your problem:
[LOG 05:38:32.044] PartLoader: Part 'BDArmory_1.3.5PR_01292022_2235/GameData/BDArmory/Parts/20mmVulcan/vulcanTurret/bahaGatlingGun' has no database record. Creating.
[LOG 05:38:32.044] [DragCubeSystem]: Drag cubes not found or cannot be read for part Part. Generating New drag cubes.
[LOG 05:38:32.054] DragCubeSystem: Creating drag cubes for part 'bahaGatlingGun'
[EXC 05:38:32.089] NullReferenceException: Object reference not set to an instance of an object
	BDArmory.Modules.ModuleWeapon.GetInfo () (at <dadbcdf9b9ad46c1a220dccbea06d76f>:0)
	PartLoader.CompilePartInfo (AvailablePart newPartInfo, Part part) (at <39c0323fb6b449a4aaf3465c00ed3c8d>:0)
	PartLoader+<CompileParts>d__56.MoveNext () (at <39c0323fb6b449a4aaf3465c00ed3c8d>:0)
	UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <12e76cd50cc64cf19e759e981cb725af>:0)
	UnityEngine.DebugLogHandler:LogException(Exception, Object)
	ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
	UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
TL;DR: Whatever "Drag cubes" are, the game can't seem to find them... and it's trying to make new ones... but it can't.

**Comment by ThatCoolCoder** [comment]
I am also having this issue and it is very annoying

**Comment by khaiqdo** [comment]
SAAAAAAAAAMEEEE AAAAAAAAAAAA

**Comment by khaiqdo** [comment]
I deleted the Vulcan from the gamedata, however, now it gets stuck on the next file, and so on.

**Comment by SuicidalInsanity** [comment]
Please confirm your install path is /GameData/BDArmory/Agencies... etc
and not /GameData/BDArmory/BDarmory/
The drag cubes the OP mentioned are not the issue - the issue is KSP's partloader not being able to find ammo definitions when loading the gun due to the BD_bullets.cfg being in the wrong place.

**Comment by Hhaha233** [comment]
> Please confirm your install path is /GameData/BDArmory/Agencies... etc and not /GameData/BDArmory/BDarmory/ The drag cubes the OP mentioned are not the issue - the issue is KSP's partloader not being able to find ammo definitions when loading the gun due to the BD_bullets.cfg being in the wrong place.

Yes, I did do it. There's not wrong about the path of BDA, but it still happens.  Even though I replaced the version and the way to install it (including the CKAN, GitHub and a package from my friend) several times, this issue hasn't changed.

**Comment by Hhaha233** [comment]
> > 请确认您的安装路径是 /GameData/BDArmory/Agency...等等，而不是 /GameData/BDArmory/BDarmory/ OP 提到的拖拽立方体不是问题 - 问题是 KSP 的零件装载机在装填枪支时无法找到弹药定义，因为BD_bullets.cfg在错误的位置。
> 
> 是的，我确实做到了。BDA的道路没有错，但它仍然会发生。即使我多次替换了版本和安装方式（包括 CKAN、GitHub 和我朋友的软件包），这个问题也没有改变。

 I found the solution. You could launch the KSP by the launcher of Private Division, but not to go straight to open the "KSP_x64.exe".
